tools/test-lib/react-app/app.test.js

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { configure, mount } from 'enzyme';
+import App, { dataSources } from './app';
+import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() });
+
+const keys = Object.keys(dataSources);
+
+describe('lib-test', () => {
+  test('renders without crashing', () => {
+    mount(<App />);
+  });
+
+  const testFirstNodeNameMatch = (wrapper, key) => {
+    const nodeNames = dataSources[key]().nodes.map(node => node.name);
+    const firstNodeName = wrapper
+      .find('.pipeline-nodelist--nested')
+      .find('NodeListRow')
+      .first()
+      .text();
+    if (key === 'random') {
+      let combinedNodeNames = [];
+      nodeNames.forEach(name => {
+        combinedNodeNames = combinedNodeNames.concat(name.split(' '));
+      });
+      firstNodeName.split(' ').forEach(word => {
+        expect(combinedNodeNames).toContain(word);
+      });
+    } else {
+      expect(nodeNames).toContain(firstNodeName);
+    }
+  };
+
+  test.each(keys)(`uses %s dataset when provided as prop`, key => {
+    const wrapper = mount(<App initialData={key} />);
+    testFirstNodeNameMatch(wrapper, key);
```

Comment:
Not required, but I think it's often a nice idea to keep the `expect` inside the test here if practical (i.e. return the two values from `testFirstNodeNameMatch`).

Reply:
Thanks! Make sense, but I think it's probably not worth the effort to change, especially given the additional complexity that it might entail. Gonna leave it for now.